### PR TITLE
Implement regex list for list adapter

### PR DESCRIPTION
**Pull request by manlinl**

### CODE REVIEW

File: mixer/adapter/list/config/config.proto [hunk 1]
```diff
@@ -58,6 +58,9 @@ message Params {
 
         // List entries are treated as IP addresses and ranges.
         IP_ADDRESSES = 2;
+
+        // List entries are treated as regex.
+        REGEX = 3;
```
Comment (apicl) [inline mixer/adapter/list/config/config.proto]: What kind of regular expression do we support here? Can you please give a few examples.

File: mixer/adapter/list/regexList.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+		}
+	}
+	return false, nil
+}
+
+func (l *regexList) numEntries() int {
+	return len(l.regexpList)
+}
+
+func parseRegexList(buf []byte, overrides []string) (*regexList, error) {
```
Comment (apicl) [inline mixer/adapter/list/regexList.go]: Please comment this function. If possible, use an example to explain.

File: mixer/adapter/list/regexList.go [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import (
+	"regexp"
+	"strings"
+)
+
+type regexList struct {
+	regexpList []*regexp.Regexp
+}
+
+func (l *regexList) checkList(symbol string) (bool, error) {
```
Comment (apicl) [inline mixer/adapter/list/regexList.go]: Please comment that the or relationship between the regular expressions.

Comment (apicl) [inline mixer/adapter/list/regexList.go]: Why do we need to return error?

File: mixer/adapter/list/config/config.proto (hunk 1 above)
Comment (manlinl) [reply mixer/adapter/list/config/config.proto]: Comment added.

File: mixer/adapter/list/regexList.go (hunk 3 above)
Comment (manlinl) [reply mixer/adapter/list/regexList.go]: done.

Comment (manlinl) [reply mixer/adapter/list/regexList.go]: checkList is a predefined interface in list.go. For other list, returning error is needed, though not for regex list.

File: mixer/adapter/list/regexList.go (hunk 2 above)
Comment (manlinl) [reply mixer/adapter/list/regexList.go]: done.

**Pull request merged**